presto-hive/src/main/java/io/prestosql/plugin/hive/InternalHiveConnectorFactory.java

```diff
@@ -133,6 +139,10 @@ public static Connector createConnector(String catalogName, Map<String, String>
                     classLoader);
             Set<Procedure> procedures = injector.getInstance(Key.get(new TypeLiteral<Set<Procedure>>() {}));
             Set<SystemTable> systemTables = injector.getInstance(Key.get(new TypeLiteral<Set<SystemTable>>() {}));
+            List<EventListener> eventListeners = injector.getInstance(Key.get(new TypeLiteral<Set<EventListener>>() {}))
+                    .stream()
+                    .map(listener -> new ClassLoaderSafeEventListener(listener, classLoader))
+                    .collect(toImmutableList());
```

Comment:
Why not to `toImmutableSet()`?

Reply:
there is not point to even attempt any deduplication

Reply:
I had two points, none of them serious:
 - `injector.getInstance(Key.get(new TypeLiteral<Set<EventListener>>() {}))` returns a set, so why do we change this?
 - Using list here could make a false impression that this could contain duplicates (set still can, but it is less obvious). I don't see a reason why one would need even thing about it.

Any way, I will it leave up to author.

Reply:
Since `ClassLoaderSafeEventListener` doesn't (and shouldn't) implement equals, using set is pointless.